server/athenian/api/db.py

```diff
@@ -391,3 +384,18 @@ async def insert_or_ignore(model,
 def extract_registered_models(base: Any) -> Mapping[str, Any]:
     """Return the mapping from declarative model names to their classes."""
     return base.registry._class_registry
+
+
+def conn_in_transaction(conn: Connection) -> bool:
+    """Return True if the connection is inside a transaction."""
+    return _conn_backend_in_transaction(conn._connection)
+
+
+def _conn_backend_in_transaction(conn_backend: ConnectionBackend) -> bool:
+    """Return True if the connection backend is inside a transaction."""
+    raw_connection = conn_backend.raw_connection
+    if isinstance(raw_connection, asyncpg.Connection):
+        return raw_connection.is_in_transaction()
+    if isinstance(raw_connection, aiosqlite.Connection):
+        return raw_connection.in_transaction
+    raise ValueError(f"Unhandled db connection type {type(raw_connection)}")
```

Comment:
this is definitely a code bug if we reach here, so let's assert:
```suggestion
    raise AssertionError(f"Unhandled db connection type {type(raw_connection)}")
```